parsers/IE.py

```diff
@@ -0,0 +1,258 @@
+from datetime import datetime, timedelta
+from logging import Logger, getLogger
+from typing import Optional
+
+import arrow
+import pytz
+from requests import Response, Session
+
+from parsers.lib.config import refetch_frequency
+from parsers.lib.exceptions import ParserException
+
+DEMAND_URL = "https://www.smartgriddashboard.com/DashboardService.svc/data?area=demandactual&region={zone}&datefrom={dt}+00%3A00&dateto={dt}+23%3A59"
+DEMAND_FORECAST_URL = "https://www.smartgriddashboard.com/DashboardService.svc/data?area=demandforecast&region={zone}&datefrom={dt}+00%3A00&dateto={dt}+23%3A59"
+WIND_URL = "https://www.smartgriddashboard.com/DashboardService.svc/data?area=windactual&region={zone}&datefrom={dt}+00%3A00&dateto={dt}+23%3A59"
+WIND_FORECAST_URL = "https://www.smartgriddashboard.com/DashboardService.svc/data?area=windforecast&region={zone}&datefrom={dt}+00%3A00&dateto={dt}+23%3A59"
+EXCHANGE_URL = "https://www.smartgriddashboard.com/DashboardService.svc/data?area=interconnection&region={zone}&datefrom={dt}+00%3A00&dateto={dt}+23%3A59"
+
+KINDS_URL = {
+    "demand": DEMAND_URL,
+    "demand_forecast": DEMAND_FORECAST_URL,
+    "wind": WIND_URL,
+    "wind_forecast": WIND_FORECAST_URL,
+    "exchange": EXCHANGE_URL,
+}
+
+ZONE_MAPPING = {
+    "IE": {"key": "ROI", "exchange": "INTER_EWIC"},
+    "GB-NIR": {"key": "NI", "exchange": "INTER_MOYLE"},
+}
+
+
+def fetch_data(
+    target_datetime: datetime,
```

Comment:
```suggestion
    target_datetime: Optional[datetime],
```
We need this to be optional since the forcast function don't set a datetime when it recives a None datetime, so it passes None on to this function.

This might require some other changes later as well to stay type safe.